### PR TITLE
Let Rust plugin fetch dependencies in pull

### PR DESCRIPTION
**Pull request by ChrisMacNaughton**
Pulling the dependencies in the pull stage allows
building Rust snaps on Launchpad builders,
which are offline during the build stage

### CODE REVIEW

**Comment by sergiusens** [comment]
Thanks for the PR, seems the unit tests need updating though.
You can try locally by running

    ./runtests.sh unit

Also might as well run

    ./runtests.sh static

Before pushing again. Thanks again

**Comment by ChrisMacNaughton** [comment]
It looks like the tests failed because travis couldn't update packages?

File: snapcraft/tests/test_plugin_rust.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def test_pull(self, run_mock, script_mock):
 
         plugin.pull()
 
-        self.assertEqual(1, run_mock.call_count)
+        self.assertEqual(2, run_mock.call_count)
```
Comment (elopio) [inline snapcraft/tests/test_plugin_rust.py]: This test would require more than checking that rust was called twice. Please take a look at the next statement that checks the arguments of the first call, and do something similar to the second.

File: snapcraft/plugins/rust.py [hunk 2]
```diff
@@ -54,6 +54,12 @@ def schema(cls):
 
     def __init__(self, name, options, project):
         super().__init__(name, options, project)
+        self.build_packages.extend([
+            'build-essential',
+            'git',
+            'curl',
+            'file',
```
Comment (ChrisMacNaughton) [inline snapcraft/plugins/rust.py]: These are the things that were required for me to get rustc installed on one of the launchpad builders. Can't find specific documentation about it but that's where this list came from.

**Comment by ChrisMacNaughton** [comment]
@elopio tests should be accurate now, and the integration rust project now pulls in a dependency :smile:

File: snapcraft/plugins/rust.py [hunk 3]
```diff
@@ -54,6 +54,12 @@ def schema(cls):
 
     def __init__(self, name, options, project):
         super().__init__(name, options, project)
+        self.build_packages.extend([
+            'build-essential',
+            'git',
```
Comment (kyrofa) [inline snapcraft/plugins/rust.py]: Git should be pulled in simply by using a git source. It shouldn't be required here.

File: snapcraft/plugins/rust.py [hunk 4]
```diff
@@ -54,6 +54,12 @@ def schema(cls):
 
     def __init__(self, name, options, project):
         super().__init__(name, options, project)
+        self.build_packages.extend([
+            'build-essential',
```
Comment (kyrofa) [inline snapcraft/plugins/rust.py]: This is a large meta package. Are you sure you need all of it?

File: snapcraft/plugins/rust.py (hunk 3 above)
Comment (kyrofa) [inline snapcraft/plugins/rust.py]: Unless the pull of rust eventually turns into a git clone. Is that the case?

Comment (ChrisMacNaughton) [inline snapcraft/plugins/rust.py]: Pulling Rust shouldn't clone, but using Cargo for the build can

File: snaps_tests/demos_tests/test_downloader_with_wiki_parts.py [hunk 5]
```diff
@@ -27,7 +27,8 @@ def test_downloader_with_wiki_parts(self):
         snap_path = self.build_snap(self.snap_content_dir)
         self.install_snap(snap_path, 'downloader', '1.0')
         expected = (
-            '.*<title>The leading OS for PC, tablet, phone and cloud '
+            '.*<title>The leading operating system for PCs, tablets, '
```
Comment (sergiusens) [inline snaps_tests/demos_tests/test_downloader_with_wiki_parts.py]: This should be reverted as it was fixed in master with a more permanent solution.

**Comment by ChrisMacNaughton** [comment]
@sergiusens I've reverted that change, and it looks like that is the issue with the autopkgtests still.

**Pull request merged**